ha/core/event_manager/event_manager.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+from cortx.utils.log import Log
+from ha.core.system_health.const import EVENTS
+from ha.core.event_manager.error import InvalidEvent
+from ha.core.event_manager.models.action_event import RecoveryActionEvent
+
+class EventManager:
+
+    __instance = None
+
+    @staticmethod
+    def get_instance():
+        """
+        Static method to fetch the current instance.
+        Performs initialization related to common database and message bus
+        """
+        if not EventManager.__instance:
+            EventManager(singleton_check=True)
+        return EventManager.__instance
+
+    def __init__(self, singleton_check: bool = False):
+        """
+        Private Constructor.
+        """
+        if singleton_check is False:
+            raise Exception("Please use EventManager.get_instance() to fetch singleton instance of class")
+        if EventManager.__instance is None:
+            EventManager.__instance = self
+        else:
+            raise Exception("EventManager is singleton class, use EventManager.get_instance().")
+
+    def _validate_component(self, component: str) -> None:
+        """
+        Validate component raise error invalid component.
+
+        Args:
+            component (str): Component name
+        """
+        pass
```

Comment:
This is still empty. We can add just hare for now.

Reply:
Yes, Here i had confusion 
should we avoid validate component to keep generic
Allow anyone to subscribe to events

1. Keep limited set
[ha, hare, csm, sspl, ..., ]

2. Keep some reserved set
[ha, hare, csm, sspl, ..., ha_res1, ha_res2, ha_res3] ha_res can be use by QA, ha test if needed

3. Allow all

Which should be good way to implement

Reply:
Keep limited set (1). Test can use existing component. I can't think of a scenario where test will need to use some reserved component like test1. If there is any use case, add one component for test.

Reply:
Updated